src/http/mod.rs

```diff
@@ -5,66 +5,148 @@ use std::{
 pub struct FormdataBody {
     pub fields: Option<Vec<FormdataText>>,
     pub files: Option<Vec<FormdataFile>>,
 }
+
+#[derive(Clone)]
 pub enum BodyType {
     Text(String),
     FormdataBody(FormdataBody),
 }
+
+#[derive(Clone)]
 pub struct Request {
     pub method: HTTPMethod,
     pub path: String,
     pub headers: HashMap<String, String>,
     pub body: Option<BodyType>,
 }
 
+impl Request {
+    pub fn new(method: HTTPMethod, path: &str, headers: HashMap<String, String>) -> Self {
+        Self {
+            method,
+            path: path.to_string(),
+            headers,
+            body: None,
+        }
+    }
+
+    /// Set body request
+    pub fn body(&mut self, body: Vec<u8>) -> &Self {
+        self.body = Some(
+            match self
+                .headers
+                .get("Content-Type")
+                .unwrap_or(&"".to_string())
+                .contains("multipart/form-data")
+            {
+                true => {
+                    self.headers.insert(
+                        "Content-Type".to_string(),
+                        "multipart/form-data".to_string(),
+                    );
+                    let formdatabody = parse_formdata(&body);
+
+                    BodyType::FormdataBody(formdatabody)
+                }
+
+                false => {
+                    let chars = std::str::from_utf8(body.as_slice()).unwrap().to_string();
+                    BodyType::Text(chars)
+                }
+            },
+        );
+
+        self
+    }
+}
+
+impl FromStr for Request {
+    type Err = RequestError;
+
+    /// Parsing http request
+    fn from_str(request: &str) -> Result<Self, Self::Err> {
+        let request_lines: Vec<&str> = request.split("\r\n").collect();
+        let mut first_line_iter = request_lines[0].split_whitespace();
+        let method = first_line_iter.next().unwrap();
+        let uri = first_line_iter.next().unwrap();
+        let mut headers: HashMap<String, String> = HashMap::new();
+        for header in request_lines.iter().skip(1) {
+            if header.len() > 0 {
+                match header.find(": ") {
+                    Some(split_index) => {
+                        headers.insert(
+                            header[..split_index].to_string(),
+                            header[split_index + 2..].to_string(),
+                        );
+                    }
+                    None => return Err(RequestError::MissingChar(':')),
+                }
+            }
+        }
+        Ok(Self {
+            path: uri.to_string(),
+            body: None,
+            method: HTTPMethod::from_str(method).unwrap(),
+            headers,
+        })
+    }
+}
+
 pub struct Response {
     pub status_code: u16,
     pub headers: Option<HashMap<String, String>>,
     pub body: Option<Vec<u8>>,
 }
 
+impl Response {
+    pub fn new(status_code: u16) -> Self {
+        Self {
+            status_code,
+            headers: None,
+            body: None,
```

Comment:
While this might be nice, I personally don't like chainable methods.

It might be easier for developers using this web server, but I feel like that is a goal for a framework built on the web server,

Reply:
I see it's more clean than old, and It's a pattern followed in most Rust projects.
In any case, The library will need to be developed eventually, and this will facilitate the development process and not only help the developers who use the library.